gems/aws-sdk-core/lib/aws-sdk-core/plugins/regional_endpoint.rb

```diff
@@ -93,6 +96,49 @@ def after_initialize(client)
         client.config.sigv4a_signing_region_set = region_set
       end
 
+      private
+
+      def initialize_default_endpoint(client)
+        client_module = Object.const_get(client.class.name.rpartition('::').first)
+        if client.config.respond_to?(:endpoint_provider) &&
+          client_module.const_defined?(:EndpointParameters) &&
+          (param_class = client_module.const_get(:EndpointParameters)) &&
+          param_class.respond_to?(:create) &&
```

Comment:
We could potentially just call create and add NoMethodError in our rescue. We could do the same with fetching endpoint_provider off config (would or would not exist) and potentially also with const_get (does this error or return nil?)